tutorials/intermediate/blitting.py

```diff
@@ -0,0 +1,186 @@
+"""
+=================
+Blitting Tutorial
+=================
+
+'Blitting' is a `standard technique
+<https://en.wikipedia.org/wiki/Bit_blit>`__ in computer graphics that
+in the context of Matplotlib can be used to (drastically) improve
+performance of interactive figures.  It is used internally by the
+:mod:`~.animation` and :mod:`~.widgets` modules for this reason.
+
+The source of the performance gains is simply not re-doing work we do
+not have to.  For example, if the limits of an Axes have not changed,
+then there is no reason we should re-draw all of the ticks and
+tick-labels (particularly because text is one of the more expensive
+things to render).
+
+The procedure to save our work is roughly:
+
+- draw the figure, but exclude any artists marked as 'animated'
+- save a copy of the Agg RBGA buffer
+
+In the future, to update the 'animated' artists we
+
+- restore our copy of the RGBA buffer
+- redraw only the animated artists
+- show the resulting image on the screen
+
+thus saving us from having to re-draw everything which is _not_
+animated.
+
+Simple Example
+--------------
+
+We can implement this via methods on `.FigureCanvasAgg` and setting
+``animated=True`` on our artist.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+x = np.linspace(0, 2 * np.pi, 100)
+
+fig, ax = plt.subplots()
+# animated=True makes the artist be excluded from normal draw tree
+(ln,) = ax.plot(x, np.sin(x), animated=True)
+
+# stop to admire our empty window axes and ensure it is drawn
```

Comment:
I very much enjoy this kind of lighthearted tone a lot in tutorials, but people arriving here from the `Animation` api docs may not know about `plt.pause()`. `plt.pause` is one of the methods in mpl whose name differs most from its purpose (here, just flushing the draw tree), and I don't think "ensure it is drawn" will be enough to cue people in who haven't seen this before into what's happening here.

```suggestion
# plt.pause is the easiest way to trigger the figure to draw all non-animated 
# artists, and let's us stop to admire our empty window axes
```

Reply:
As documented in https://github.com/matplotlib/matplotlib/pull/4779 (which I'm also trying to get across the finish line!).  If you think of `plt.pause` as primarily suspending the execution of your script (by running the event loop) `pause` does make sense.

I'll re-word this a bit, bu I think that this suggestion is a bit misleading in other ways (as `plt.pause` will only trigger a draw on the current figure if it is stale)

Reply:
I assume it makes good sense given the correct mental model of the event loop. Unfortunately, it does not make intuitive sense IME to people (like me) who don't know how the event loop is, and just want to have a way to do what "plt.show" sounds like it does, which is what it feels like you're doing here.

Please feel free to change the wording to more appropriately describe how one should be thinking about what "plt.pause" is doing here. My top-level suggestion is that, if possible, someone who is new to mpl and reading this (and likely has no mental model for the event loop) should be able to understand what `plt.pause` is doing in this context.

Reply:
Added a `plt.show(block=False)` and tweak the comments here a bit.